### DISCUSSION TITLE
semp.gateway.address als ENV-Variable unter Docker

### DISCUSSION THREAD

**Discussion by electric-m** [camueller/SmartApplianceEnabler#232] [Installation und Betrieb]
Hallo zusammen,

kann die semp.gateway.address auch über die Docker ENV-Variables definiert werden?

name: semp.gateway.address
value: Die IP des SAE?

Viele Grüße

edit: habe es jetzt über die Shell im Container eingetragen. Der Home Manager findet den SAE trotzdem nicht.

unter IP/semp sehe ich jedoch die Daten.
Multicast sollte sich funktionieren.

**Comment by camueller** [comment]
Steht in der Doku: https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Docker_DE.md#starten-des-containers-2

**Reply by electric-m** [reply]
Ich konnte die Variable ja gestern schon setzen.
Die Funktionsprüfung, die in der Doku zu SEMP beschrieben werden, habe ich durchgeführt, und erhalte unter saeIP/semp diese Ausgabe:

`F-11121985-000000000001-00 Vitocal 333-G-NC HeatPump 7537968501007103 Viessmann 3060 Measurement false false false F-11121985-000000000001-00 false Off 0 0 60`

Möglicherweise liegt es daran, dass die Verbindung zu meinem Shelly 3EM noch nicht steht? (WLAN Empfang ist aktuell noch zu schwach, muss mir einen Repeater besorgen).

ich dachte nur, da das Gerät ja angelegt ist, dass es sich über SEMP bekannt gegeben wird.

**Comment by electric-m** [comment]
Also, nur als Feedback,

nachdem alle Fehlerbehebungen gescheitert sind, habe ich nun für den Container ein macvlan konfiguriert.
Nun funktioniert alles!

Ich hatte nur gedacht, dass, wie in der Doku beschrieben, auch host als Network funktioniert.

Viele Grüße

**Comment by pkuehnel** [comment]
Ich habe das selbe Problem, möchte eigtl gerne über Host zugreifen. `/semp` funktioniert, aber über wireshark kommen keine UDP Pakete von der IP Adresse, ich vermute er nutzt ein anderes Interface. Wenn ich nach Doku vorgehe (ich nutze die docker-compose.yml) komme ich nicht auf die Weboberfläche drauf, wenn ich netzwork-mode: host nutze, kommen die UDP Pakete nicht.

Edit: Eigtl möchte ich den Network mode Host nutzen, ich habe in meinem kompletten Netzwerk keine Geräte bei denen eine feste IP eingetragen ist, sondern nur alles über den DHCP Server als Reservierung.
